### PR TITLE
[react-native-auth0] Fixed Generics, and expanded on types

### PR DESCRIPTION
**Pull request by Wolfleader101**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change. (Tests did not need to be changed, as it was just updating types)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

It should be noted, it also changes the `user` in `Auth0ContextInterface` to correctly match the Auth0 user returned from their API.

### CODE REVIEW

**Comment by Wolfleader101** [comment]
Not really sure how to fix this - its kind of required so that I can pass it to the user type...

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-native-auth0 ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-native-auth0@latest/))
was missing the following properties:
1. TimeoutError
1. LocalAuthenticationStrategy
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against e2f2644f9d0d0ca138c4164cebe75c2fa2b37813
</p>

**Comment by gabritto** [comment]
I'd like to wait for an owner review on this. In the meantime, please consider updating the tests.

File: types/react-native-auth0/index.d.ts [hunk 1]
```diff
@@ -182,39 +180,56 @@ export class Auth {
     multifactorChallenge(params: MultiFactorChallengeParams): Promise<MultiFactorChallengeResponse>;
 }
 
+export interface Auth0Connection {
```
Comment (poovamraj) [inline types/react-native-auth0/index.d.ts:183]: Identities wouldn't necessarily just include this structure, You can check the example here - https://auth0.com/docs/manage-users/user-accounts/user-account-linking

File: types/react-native-auth0/index.d.ts [hunk 2]
```diff
@@ -182,39 +180,56 @@ export class Auth {
...
  */
-export interface Auth0User<T> {
+export interface Auth0User<T = void> {
     created_at: string;
-    email: string;
-    emailVerified: boolean;
-    identities: any[];
+    email?: string;
+    email_verified?: boolean;
+    identities: Auth0Connection[];
```
Comment (poovamraj) [inline types/react-native-auth0/index.d.ts:197]: As mentioned above, the identities structure wouldn't have particular definition.

File: types/react-native-auth0/index.d.ts [hunk 3]
```diff
@@ -182,39 +180,56 @@ export class Auth {
...
+    name?: string;
+    given_name?: string;
+    family_name?: string;
+    middle_name?: string;
+    nickname?: string;
     picture?: string;
     updated_at: string;
-    userId: string;
+    user_id: string;
+    sub: string;
```
Comment (poovamraj) [inline types/react-native-auth0/index.d.ts:209]: Any reference that this will be non null?

File: types/react-native-auth0/index.d.ts [hunk 4]
```diff
@@ -312,8 +327,8 @@ export default class Auth0 {
     users(token: string): Users;
 }
 
-export class Auth0ContextInterface {
-    user: any;
+export interface Auth0ContextInterface<T = void> {
```
Comment (poovamraj) [inline types/react-native-auth0/index.d.ts:330]: defaulting generics to void is nice one 👍

**Pull request closed without merging**